packages/core/src/controller/object-controller.ts

```diff
@@ -0,0 +1,51 @@
+import { Task } from '../task';
+import type { Controller } from './controller';
+import { createFuture } from '../future';
+import { OperationObject } from '../operation';
+import { isObjectOperation } from '../predicates';
+import { extractLabels } from './utils';
+
+export function createObjectController<TOut>(task: Task<TOut>, obj: OperationObject<TOut>, createController: () => Controller<TOut>): Controller<TOut> {
```

Comment:
`ObjectController` and `FunctionController` are basically the same? Can we make a `DelegateController`?